src/AutoPilotPlugins/APM/APMSensorsComponent.qml

```diff
@@ -327,23 +341,54 @@ SetupPage {
                     visible: sensorParams.rgCompassAvailable[index]
 
                     QGCLabel {
-                        text: qsTr("Compass ") + (index+1) + " " +
-                              (sensorParams.rgCompassPrimary[index] ? qsTr("(primary") :qsTr( "(secondary")) +
-                              (sensorParams.rgCompassExternalParamAvailable[index] ?
-                                   (sensorParams.rgCompassExternal[index] ? qsTr(", external") : qsTr(", internal") ) :
-                                   "") +
-                              ")"
+                        text: compassLabel(index)
                     }
 
                     Column {
                         anchors.margins:    ScreenTools.defaultFontPixelWidth * 2
                         anchors.left:       parent.left
                         spacing:            Math.round(ScreenTools.defaultFontPixelHeight / 4)
 
-                        FactCheckBox {
-                            text:       qsTr("Use Compass")
-                            fact:       sensorParams.rgCompassUseFact[index]
-                            visible:    sensorParams.rgCompassUseParamAvailable[index] && !sensorParams.rgCompassPrimary[index]
+                        RowLayout {
+                            spacing: ScreenTools.defaultFontPixelWidth
+
+                            FactCheckBox {
+                                id:         useCompassCheckBox
+                                text:       qsTr("Use Compass")
+                                fact:       sensorParams.rgCompassUseFact[index]
+                                visible:    sensorParams.rgCompassUseParamAvailable[index] && !sensorParams.rgCompassPrimary[index]
+                            }
+
+                            QGCComboBox {
+                                model:      [ qsTr("Priority 1"), qsTr("Priority 2"), qsTr("Priority 3"), qsTr("Not Set") ]
+                                visible:    _singleCompassSettingsComponentShowPriority && sensorParams.compassPrioFactsAvailable && useCompassCheckBox.visible && useCompassCheckBox.checked
+
+                                property int _compassIndex: index
+
+                                function selectPriorityfromParams() {
+                                    if (visible) {
+                                        currentIndex = 3
```

Comment:
I think this function always selects the last index item before cycling through and finding the right one. Would it make more sense to put this after the for loop? It's probably not noticeable in the UI.

Reply:
@ptran TBH this is just a cherry-pick from upstream QGC, so I don't have the answers to your questions. But, appears to work on the bench...